src/libs/SearchUtils.ts

```diff
@@ -542,29 +545,67 @@ function getPolicyIDFromSearchQuery(queryJSON: SearchQueryJSON) {
     return policyID;
 }
 
+function getDisplayValue(
+    filterName: string,
+    filter: string,
+    personalDetails: OnyxTypes.PersonalDetailsList,
+    cardList: OnyxTypes.CardList,
+    reports: OnyxCollection<OnyxTypes.Report>,
+    taxRates: Record<string, string[]>,
+) {
+    if (filterName === CONST.SEARCH.SYNTAX_FILTER_KEYS.FROM || filterName === CONST.SEARCH.SYNTAX_FILTER_KEYS.TO) {
+        return PersonalDetailsUtils.createDisplayName(personalDetails[filter]?.login ?? '', personalDetails[filter]);
+    }
+    if (filterName === CONST.SEARCH.SYNTAX_FILTER_KEYS.CARD_ID) {
+        return cardList[filter].bank;
+    }
+    if (filterName === CONST.SEARCH.SYNTAX_FILTER_KEYS.IN) {
+        return ReportUtils.getReportName(reports?.[`${ONYXKEYS.COLLECTION.REPORT}${filter}`]);
+    }
+    if (filterName === CONST.SEARCH.SYNTAX_FILTER_KEYS.TAX_RATE) {
+        const result: string[] = [];
+        Object.entries(taxRates).forEach(([taxRateName, taxRateKeys]) => {
+            if (!taxRateKeys.some((taxRateKey) => taxRateKey === filter) || result.includes(taxRateName)) {
+                return;
+            }
+            result.push(taxRateName);
```

Comment:
instead of `forEach` + assigning to `results` array, I think you can do either `.map()` or `map() + filter()` and this will look nicer